src/marqo/s2_inference/s2_inference.py

```diff
@@ -43,7 +46,18 @@ def vectorise(model_name: str, content: Union[str, List[str]], model_properties:
     _update_available_models(model_cache_key, model_name, validated_model_properties, device, normalize_embeddings)
 
     try:
-        vectorised = available_models[model_cache_key].encode(content, normalize=normalize_embeddings, **kwargs)
+        if isinstance(content, str):
+            vectorised = available_models[model_cache_key].encode(content, normalize=normalize_embeddings, **kwargs)
+        else:
+            vector_batches = []
+            batch_size = read_env_vars_and_defaults(EnvVars.MARQO_MAX_VECTORISE_BATCH_SIZE)
+            for batch in generate_batches(content, batch_size=batch_size):
+                vector_batches.append(available_models[model_cache_key].encode(batch, normalize=normalize_embeddings, **kwargs))
+            if not vector_batches or all(
```

Comment:
Under what circumstance should we return an empty array instead of raising an error here?

Reply:
_convert_vectorized_output will actually raise an error if it's given an empty batch. But it is best to explicitly raise the error.

Reply:
done.